lib/Feature/FeatureModelTransaction.cpp

```diff
@@ -16,6 +16,27 @@ void addFeature(FeatureModel &FM, std::unique_ptr<Feature> NewFeature,
   Trans.commit();
 }
 
+void removeFeature(FeatureModel &FM, Feature *OldFeature, bool Recursive) {
+  auto Trans = FeatureModelModifyTransaction::openTransaction(FM);
+  detail::FeatureVariantTy F = detail::FeatureVariantTy(OldFeature);
+  Trans.removeFeature(F, Recursive);
+  Trans.commit();
+}
+
+void addRelationship(FeatureModel &FM, Feature *GroupRoot,
+                     Relationship::RelationshipKind Kind) {
+  auto Trans = FeatureModelModifyTransaction::openTransaction(FM);
+  detail::FeatureVariantTy GR = detail::FeatureVariantTy(GroupRoot);
```

Comment:
Can you already provide the variant as a param to the function?

Reply:
We could do that. However, if we do, we should use them consistently in all helper functions.
Originally, the variants were introduced to reference a feature for which you could not provide a pointer. Here, the caller can always get the pointer with FM.getFeature(name).

Reply:
That's a good point, I guess if the user has a FM it is better to always require a pointer. Supporting a lookup with just a given name would mean we have to deal with the error handling if it is not present. If we only take a feature we can make sure the user lookup the feature and handled what ever problems he had if he did not find one for his name.

Reply:
Decided in meeting to use references instead of pointers of FeatureVariant.